homeassistant/components/ecobee/sensor.py

```diff
@@ -114,13 +137,43 @@ def state(self):
         if self.type == "temperature":
             return float(self._state) / 10
 
+        if self.type == NOTIFICATIONS_KEY:
+            return len(
+                [
+                    notification
+                    for notification in self._attributes
+                    if notification["enabled"]
+                    and datetime.strptime(notification["remindMeDate"], "%Y-%m-%d")
+                    <= datetime.now()
+                ]
+            )
+
         return self._state
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement this sensor expresses itself in."""
         return self._unit_of_measurement
 
+    @property
+    def state_attributes(self):
+        """Return the attributes of the sensor."""
+        if self.type != NOTIFICATIONS_KEY:
+            return None
+
+        if self._attributes is None:
+            self._attributes = {}
+        else:
+            self._attributes = {
```

Comment:
We don't need to create a new dict if we just want to update one key. Just set the new value for that key in the existing dict.